src/Provider/Keycloak.php

```diff
@@ -68,11 +83,67 @@ public function __construct(array $options = [], array $collaborators = [])
     }
 
     /**
+     * We need to cache the access token locally allowing for later optional post-processing
+     * by `checkForKeycloakRoles()`
+     *
+     * @param mixed $grant
+     * @param array $options
+     * @return AccessToken
+     */
+    public function getAccessToken($grant, array $options = [])
+    {
+        $this->accessToken = parent::getAccessToken($grant, $options);
+        return $this->accessToken;
+    }
+
+    /**
+     * Check for Keycloak-supplied additional fields held by the access token which in turn is inside accessToken.
+     *
+     * @return KeyCloakRoles
+     */
+    public function getKeycloakRoles()
```

Comment:
This functionality seems valuable and I don't think it should be so tightly coupled with access token workflow. I honestly feel like it should not be part of this project, and if it is part of the package, I would prefer it be made available as a static class for tools; in a `Stevenmaguire\OAuth2\Client\Tools` namespace. 

```php
public static function getKeycloakRoles(League\OAuth2\Client\Token\AccessToken $accessToken, $encryptionKey, $encryptionAlgorithm)
{
    //
}
```